examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+            // looking at the graphical output:
+            else
+              {
+                for (unsigned int dim_idx = 0; dim_idx < dim; ++dim_idx)
+                  {
+                    force[dim_idx](cell->active_cell_index()) =
+                      parameters.right_hand_side.value(cell->center(), dim_idx);
+                    pml[dim_idx](cell->active_cell_index()) = -1e+20;
+                  }
+              }
+          }
+
+        for (unsigned int dim_idx = 0; dim_idx < dim; ++dim_idx)
+          {
+            data_out.add_data_vector(force[dim_idx],
+                                     "force_" + std::to_string(dim_idx));
+            data_out.add_data_vector(pml[dim_idx],
+                                     "pml_" + std::to_string(dim_idx));
+          }
+        data_out.add_data_vector(rho, "rho");
+
+        data_out.build_patches();
+
+        unsigned int      nb_number_positions;
+        std::stringstream frequency_idx_stream;
+        nb_number_positions =
+          ((unsigned int)std::log10(parameters.nb_frequency_points)) + 1;
+        frequency_idx_stream << std::setw(nb_number_positions)
+                             << std::setfill('0') << frequency_idx;
+        std::string filename = (parameters.simulation_name + "_" +
+                                frequency_idx_stream.str() + ".vtu");
+        data_out.write_vtu_in_parallel(filename.c_str(), mpi_communicator);
+      }
+  }
+
+
+
+  // @sect4{ElasticWave::output_results}
+
+  // This function writes the datasets that have not already been written.
+  template <int dim>
+  void ElasticWave<dim>::output_results()
+  {
+    // The vectors `frequency` and `position` are the same for all the
+    // processes. Therefore any of the processes can write the corresponding
+    // `datasets`. Because the call HDF5::DataSet::write is MPI collective, the
+    // rest of the processes will have to call HDF5::DataSet::write_none.
+    if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+      {
+        frequency_dataset.write(frequency);
+        position_dataset.write(position);
+      }
+    else
+      {
+        frequency_dataset.write_none<double>();
+        position_dataset.write_none<double>();
+      }
+  }
+
+
+
+  // @sect4{ElasticWave::setup_quadrature_cache}
+
+  // We use this function at the beginning of our computations to set up initial
+  // values of the cache variables. This function has been described in step-18.
+  // There are no differences with the function of step-18.
+  template <int dim>
+  void ElasticWave<dim>::setup_quadrature_cache()
+  {
+    unsigned int our_cells = 0;
+    for (typename Triangulation<dim>::active_cell_iterator cell =
+           triangulation.begin_active();
+         cell != triangulation.end();
+         ++cell)
+      if (cell->is_locally_owned())
+        ++our_cells;
+
+    triangulation.clear_user_data();
+
+    {
+      std::vector<QuadratureCache<dim>> tmp;
+      tmp.swap(quadrature_cache);
```

Comment:
I had to do a double-take on this. I think it would be easier to read as
```suggestion
  quadrature_cache.swap (tmp);
```
If you feel like it, feel free to post a similar patch for step-18 :-)

Reply:
I'll submit a PR!